src/System.IO.FileSystem/src/System/IO/Directory.cs

```diff
@@ -563,6 +563,9 @@ public static void Move(String sourceDirName, String destDirName)
             String destinationRoot = Path.GetPathRoot(destPath);
             if (!String.Equals(sourceRoot, destinationRoot, pathComparison))
                 throw new IOException(SR.IO_SourceDestMustHaveSameRoot);
+            
+            if (FileSystem.Current.DirectoryExists(fulldestDirName))
+                throw new IOException(SR.Format(SR.IO_AlreadyExists_Name, fulldestDirName));
```

Comment:
In this particular case I don't think it is a huge deal, as there won't be data loss with an empty directory (our helpful check to get close in behavior is ok). I wonder about File.Move's behavior though...